nselib/httpcookies.lua

```diff
@@ -0,0 +1,534 @@
+  table.insert(cookiejar, cookie3)
+
+  local cookie = CookieJar:new(cookiejar)
+
+  --Tests for new and parse function
+  test_suite:add_test(unittest.identical(cookie.cookies[1], cookie1), "Parsing of cookie1 checked")
+  test_suite:add_test(unittest.identical(cookie.cookies[2], cookie2), "Parsing of cookie2 checked")
+  test_suite:add_test(unittest.identical(cookie.cookies[3], cookie3), "Parsing of cookie3 checked")
+
+  --Test for add cookie function
+  local cookie4 = {
+    name = "PHP_SESSIONID",
+    value = "cp392d294j9dm",
+    domain = "google.subdomain.com"
+  }
+
+  cookie:add_cookie(cookie4)
+  test_suite:add_test(unittest.identical(cookie.cookies[4], cookie4), "Parsing of cookie4 checked")
+
+  --Test for update_cookie function
+  local cookie2_update = {
+    name = "SID",
+    value = "high",
+    ["max-age"] = "1200",
+  domain = "google.com",
+   }
+
+  cookie:update_cookie(cookie2_update) 
+  test_suite:add_test(unittest.identical(cookie2_update, cookie2_update), "Update cookie function verified")
+
+  --Test for get_cookie function
+  local status, c = cookie:get_cookie("session_id")
+  test_suite:add_test(unittest.identical(c, cookie.cookies[3]), "get_value function verified")
+  status, c = cookie:get_cookie("wrong_value")
+  test_suite:add_test(unittest.is_false(status), "get_value function  verified")
+
+  -- Test for no_cookie override function
+  cookie:set_no_cookie_overwrite(true)
+  test_suite:add_test(unittest.is_true(cookie.options.no_cookie_overwrite), "no_cookie_overwrite is verified")
+
+  --Test for merge function
+  --The cookie shouldnt change because we have set no_cookie_overwrite as true.
+  local cookie6 = {
+      name = "SESSIONID",
+      value = "Pdkjn94jk0d",
+      secure = true,
+      httponly = true,
+  }
+
+  cookie:merge_cookie_table({cookie6})
+  test_suite:add_test(unittest.identical(cookie.cookies[1], cookie1), "merge_cookie_table with no_cookie_overwrite verified")
+
+  cookie:set_no_cookie_overwrite(false)
+  cookie:merge_cookie_table({cookie6})
+
+  test_suite:add_test(unittest.identical(cookie.cookies[1], cookie6), "merge_cookie_table with no_cookie_overwrite verified")
+
+  local cookie7 = {
+    name = 'myCookie',
+    value = 'eaea',
+    expires = "Thu, 18 Dec 2013 12:00:00 GMT"
+  }
+
+  local cookie8 = {
+    name = "SessionI",
+    value="84nf4m0",
+    path='/search'
+  }
+
+  cookie:merge_cookie_table({cookie7,cookie8})
+  test_suite:add_test(unittest.identical(cookie.cookies[5], cookie7), "merge_cookie_table function verified")
+  test_suite:add_test(unittest.identical(cookie.cookies[6], cookie8), "merge_cookie_table function verified")
+
+  local cookiejar = cookie:check_cookie_attributes("www.google.com", 80, '/')
+  --Now we have cookie6, cookie2_update, cookie3, cookie4, cookie7 and cookie8.
+  --CookieJar shouldnt have cookie7(expires attribute)
+  --Checked expires, path, domain, secure attribute.
+
+  local cookiejar1 = {}
+
+  table.insert(cookiejar1, cookie6)
```

Comment:
Actually, I am not using the library here and I just want to add the tables inside the table. So I think its best to use table.insert here

Reply:
The question is, why are you not using the library? It will make the calls consistent and since you are providing the interface to add cookies anyway.